CMakeLists.txt

```diff
@@ -875,28 +844,32 @@ add_custom_target(clean-all
 ########################################
 # Tests                                #
 ########################################
-add_executable(urltest test/url.cpp ${BINARY_DIR}/mist/.headers)
+add_executable(urltest test/url.cpp)
 target_link_libraries(urltest mist)
-add_test(URLTest COMMAND urltest)
-add_executable(logtest test/log.cpp ${BINARY_DIR}/mist/.headers)
+add_executable(logtest test/log.cpp)
 target_link_libraries(logtest mist)
-add_test(LOGTest COMMAND logtest)
-add_executable(downloadertest test/downloader.cpp ${BINARY_DIR}/mist/.headers)
+add_executable(downloadertest test/downloader.cpp)
 target_link_libraries(downloadertest mist)
-add_test(DownloaderTest COMMAND downloadertest)
-add_executable(urireadertest test/urireader.cpp ${BINARY_DIR}/mist/.headers)
+add_executable(urireadertest test/urireader.cpp)
 target_link_libraries(urireadertest mist)
-add_test(URIReaderTest COMMAND urireadertest)
-add_executable(jsontest test/json.cpp ${BINARY_DIR}/mist/.headers)
+add_test(URIReaderTest urireadertest)
+add_executable(jsontest test/json.cpp)
 target_link_libraries(jsontest mist)
-add_test(JSONTest COMMAND jsontest)
-add_executable(resolvetest test/resolve.cpp ${BINARY_DIR}/mist/.headers)
+add_executable(resolvetest test/resolve.cpp)
 target_link_libraries(resolvetest mist)
-add_executable(bitwritertest test/bitwriter.cpp ${BINARY_DIR}/mist/.headers)
+add_executable(bitwritertest test/bitwriter.cpp)
 target_link_libraries(bitwritertest mist)
-add_test(BitWriterTest COMMAND bitwritertest)
-add_executable(streamstatustest test/status.cpp ${BINARY_DIR}/mist/.headers)
+add_test(BitWriterTest bitwritertest)
+add_executable(streamstatustest test/status.cpp)
 target_link_libraries(streamstatustest mist)
-add_executable(websockettest test/websocket.cpp ${BINARY_DIR}/mist/.headers)
+add_executable(websockettest test/websocket.cpp)
 target_link_libraries(websockettest mist)
 
+# logtest & jsontest hangs
```

Comment:
This was curious.  The CI isnt very effective if most of the tests are not working.

Are these tests working for other people?

Should we create issues about these tests?

Reply:
Most test binaries were made to allow scripted batch testing (and thus operate on stdin/stdout). @stronk-dev did some work last year on these scripts - it may be worthwhile to dig that work up again and see how much test coverage it all provides. 🤔